File: sols/bootcamp_c/quest04/ex05/my_spaceship.c
#include <stdlib.h>
#include <string.h>
#include <stdio.h>
char* my_spaceship(char* param_1)
{
    int x = 0;
    int y = 0;
    char* drctn = "up";
    

    for(int i = 0; i < strlen(param_1); i++) {

        if(param_1[i] =='A') {
            if(strcmp(drctn, "up") == 0){
                y--;
            }
            else if(strcmp(drctn, "down") == 0){
                y++;
            }
            else if(strcmp(drctn, "right") == 0){
                x++;
            }
            else if(strcmp(drctn, "right") == 0){
                x--;
            }
        }

        else if(param_1[i] == 'R') {
            if(strcmp(drctn, "up") == 0){                
                drctn = malloc(6);
                strcpy(drctn,"right");
            }
            else if(strcmp(drctn, "right") == 0){
                drctn = malloc(5);
                strcpy(drctn,"down");
            }
            else if(strcmp(drctn, "down") == 0){
                drctn = malloc(5);
                strcpy(drctn,"left");
            }   
            else if(strcmp(drctn, "left") == 0){
                drctn = malloc(3);
                strcpy(drctn,"up");
            }             
        }

        else if(param_1[i] == 'L') {
            if(strcmp(drctn, "right") == 0){
                drctn = malloc(3);
                strcpy(drctn,"up");
            }
            else if(strcmp(drctn, "up") == 0){
                drctn = malloc(5);
                strcpy(drctn,"left");
            }
            else if(strcmp(drctn, "left") == 0){
                drctn = malloc(5);
                strcpy(drctn,"down");
            }   
            else if(strcmp(drctn, "down") == 0){
                drctn = malloc(6);
                strcpy(drctn,"right");
            }
        }
    }

    char* result = malloc(sizeof(char)*(23+strlen(drctn))+sizeof(int)*2);
    sprintf(result,"{x: %d, y: %d, direction: '%s'}", x, y, drctn);
    return result;
}

File: sols/bootcamp_js/my_spaceship/ex00/my_spaceship.js
function my_spaceship(pos_ch) {
    
    var x = 0;
    var y = 0;
    var moveStr = "";
    var drc = "up";
    
    for(var i=0; i<pos_ch.length; i++){
        if (pos_ch[i] == 'A'){
            if (drc == "up") {
                y--;
            }
            else if (drc == "down") {
                y++;
            }
            else if (drc == "left") {
                x--;
            }
            else if (drc == "right") {
                x++;
            }
        }
        if (pos_ch[i] == 'R'){
            if (drc == "up") {
                drc = "right";
            }
            else if (drc == "right") {
                drc = "down";
            }
            else if (drc == "down") {
                drc = "left";
            }
            else if (drc == "left") {
                drc = "up";
            }
        }
        if (pos_ch[i] == 'L'){
            if (drc == "up") {
                drc = "left";
            }
            else if (drc == "left") {
                drc = "down";
            }
            else if (drc == "down") {
                drc = "right";
            }
            else if (drc == "right") {
                drc = "up";
            }
        }
    }
    
    //x: 2, y: -1, direction: 'down'
    moveStr = '{'+ "x: "+x+", y: "+y+", direction: '"+drc+"'"+'}';
    return moveStr;
}
my_spaceship("RAALALL")